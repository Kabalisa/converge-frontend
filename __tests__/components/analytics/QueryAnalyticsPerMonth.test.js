import React from 'react';
import wait from 'waait';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { MockedProvider } from 'react-apollo/test-utils';
import QueryAnalyticsPerMonth, { QueryAnalyticsPerMonthPagination } from '../../../src/components/analytics/QueryAnalyticsPerMonth';
import ANALYTICS_MEETING_ROOM_PER_MONTH from '../../../src/graphql/queries/analytics';


describe('QueryAnalyticsPerMonth component', () => {
  it('should render without error', () => {
    renderer.create((
      <MockedProvider mocks={[]}>
        <QueryAnalyticsPerMonth />
      </MockedProvider>
    ));
  });

  it('should render analytics data', async () => {
    const analyticsMock = {
      request: { query: ANALYTICS_MEETING_ROOM_PER_MONTH },
      result: {
        data: {
          monthlyDurationsOfMeetings: {
            MeetingsDurationaAnalytics: [
              {
                roomName: 'Tortuga',
                count: 1,
                totalDuration: 61,
              },
            ],
          },
        },
      },
    };

    const component = renderer.create((
      <MockedProvider mocks={[analyticsMock]} addTypename={false}>
        <QueryAnalyticsPerMonth />
      </MockedProvider>
    ));

    await wait(0); // wait for response

    const tree = component.toJSON();
    expect(tree.children).toHaveLength(3);
  });
});

describe('QueryAnalyticsPerMonthPagination component', () => {
  const paginationWrapper = mount(<QueryAnalyticsPerMonthPagination />);
  it('should click next', () => {
    paginationWrapper.find('#next').at(0).simulate('click');
    expect(paginationWrapper).toMatchSnapshot();
  });
});

