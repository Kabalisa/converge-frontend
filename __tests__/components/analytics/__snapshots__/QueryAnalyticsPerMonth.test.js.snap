// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryAnalyticsPerMonthPagination component should click next 1`] = `
<QueryAnalyticsPerMonthPagination>
  <Pagination
    handleData={[Function]}
    hasNext={true}
    hasPrevious={false}
    itemsPerPage={
      Array [
        5,
        10,
        20,
        50,
      ]
    }
    reverse={true}
    totalPages={50}
  >
    <nav
      className="page-navigation"
    >
      <div
        className="perPageBlock"
      >
        <ul
          className="pagination"
        >
          <select
            className="page-select"
            name="perPage"
            onChange={[Function]}
            value={5}
          >
            <option
              key="6"
            >
              5
            </option>
            <option
              key="11"
            >
              10
            </option>
            <option
              key="21"
            >
              20
            </option>
            <option
              key="51"
            >
              50
            </option>
          </select>
          <li>
            Items per page
          </li>
        </ul>
      </div>
      <div
        className="showingBlock"
      >
        <ul
          className="pagination"
        >
          <li>
            <button
              className="disabled"
              id="previous"
              onClick={[Function]}
            >
              Previous
            </button>
          </li>
          <li
            className="results"
          >
            <ul>
              <span>
                Showing 
              </span>
              <select
                className="totalPage"
                name="page"
                onChange={[Function]}
                value={2}
              >
                <option
                  key="0"
                >
                  1
                </option>
                <option
                  key="1"
                >
                  2
                </option>
                <option
                  key="2"
                >
                  3
                </option>
                <option
                  key="3"
                >
                  4
                </option>
                <option
                  key="4"
                >
                  5
                </option>
                <option
                  key="5"
                >
                  6
                </option>
                <option
                  key="6"
                >
                  7
                </option>
                <option
                  key="7"
                >
                  8
                </option>
                <option
                  key="8"
                >
                  9
                </option>
                <option
                  key="9"
                >
                  10
                </option>
                <option
                  key="10"
                >
                  11
                </option>
                <option
                  key="11"
                >
                  12
                </option>
                <option
                  key="12"
                >
                  13
                </option>
                <option
                  key="13"
                >
                  14
                </option>
                <option
                  key="14"
                >
                  15
                </option>
                <option
                  key="15"
                >
                  16
                </option>
                <option
                  key="16"
                >
                  17
                </option>
                <option
                  key="17"
                >
                  18
                </option>
                <option
                  key="18"
                >
                  19
                </option>
                <option
                  key="19"
                >
                  20
                </option>
                <option
                  key="20"
                >
                  21
                </option>
                <option
                  key="21"
                >
                  22
                </option>
                <option
                  key="22"
                >
                  23
                </option>
                <option
                  key="23"
                >
                  24
                </option>
                <option
                  key="24"
                >
                  25
                </option>
                <option
                  key="25"
                >
                  26
                </option>
                <option
                  key="26"
                >
                  27
                </option>
                <option
                  key="27"
                >
                  28
                </option>
                <option
                  key="28"
                >
                  29
                </option>
                <option
                  key="29"
                >
                  30
                </option>
                <option
                  key="30"
                >
                  31
                </option>
                <option
                  key="31"
                >
                  32
                </option>
                <option
                  key="32"
                >
                  33
                </option>
                <option
                  key="33"
                >
                  34
                </option>
                <option
                  key="34"
                >
                  35
                </option>
                <option
                  key="35"
                >
                  36
                </option>
                <option
                  key="36"
                >
                  37
                </option>
                <option
                  key="37"
                >
                  38
                </option>
                <option
                  key="38"
                >
                  39
                </option>
                <option
                  key="39"
                >
                  40
                </option>
                <option
                  key="40"
                >
                  41
                </option>
                <option
                  key="41"
                >
                  42
                </option>
                <option
                  key="42"
                >
                  43
                </option>
                <option
                  key="43"
                >
                  44
                </option>
                <option
                  key="44"
                >
                  45
                </option>
                <option
                  key="45"
                >
                  46
                </option>
                <option
                  key="46"
                >
                  47
                </option>
                <option
                  key="47"
                >
                  48
                </option>
                <option
                  key="48"
                >
                  49
                </option>
                <option
                  key="49"
                >
                  50
                </option>
              </select>
              <span
                className="pageNum reverse"
              >
                 of 
                50
                 Results
              </span>
            </ul>
          </li>
          <li>
            <button
              className="enabled"
              id="next"
              onClick={[Function]}
            >
              Next
            </button>
          </li>
        </ul>
      </div>
    </nav>
  </Pagination>
</QueryAnalyticsPerMonthPagination>
`;
